[IMP] Taxes: improve tax configuration doc (#15059)
odoo/documentation: content/applications/finance/accounting/taxes.rst

Taxes
=====

Odoo makes it easy to keep up with tax obligations, whether it's correctly quoting tax amounts on
sales, keeping track of tax debit and credit, or completing tax returns.

Overview of the tax flow
========================

In a nutshell, the flow for managing taxes within Odoo goes like this:

1. You add taxes on documents created via the Sale, Purchase and Accounting and Point of Sale apps.
2. Odoo automatically computes tax amounts.
3. On accounting documents, Odoo generates journal items to keep track of tax debit and tax credit.
4. You can view the amounts billed and taxed in the Tax Return report and use them to file a tax

>    return.

Reviewer: Why "billed"? What about "You can view the incoming and outgoing taxes charged on invoices and bills..."?

The base and tax amounts billed for each tax over a given period can be viewed in the
:doc:`Tax Return <reporting/tax_returns>` report.

.. image:: taxes/tax-return.png
   :alt: The Tax Return report shows the base and tax amounts for each tax.

.. _taxes/configuration:

Basic tax configuration
=======================

The following basic steps will set up taxes for production use in Odoo.

1. Enable any relevant :ref:`company-wide options <taxes/configuration/company>`
2. Activate any needed :ref:`pre-configured taxes <taxes/list_activation>`
3. Assign taxes on your :ref:`products <taxes/product>`

.. _taxes/configuration/company:

Company-wide options
--------------------

To access these configuration options, go to :menuselection:`Accounting --> Configuration
--> Settings` and scroll down to :guilabel:`Taxes`.

.. _taxes/default:

Default taxes
~~~~~~~~~~~~~

The default :guilabel:`Sales Tax` and :guilabel:`Purchase Tax` are automatically set on products
when creating new products.

If :ref:`Accounting Firms <accounting/fiduciaries>` mode is enabled, the default sales tax is
automatically set on new invoice lines, and the default purchase tax is automatically set on new

> vendor bill lines.

Reviewer: What do you mean by "new" invoice or vendor bill lines? The default sales/purchase taxes aren't used on invoice/bill lines without the Accounting Firms mode being set? That seems odd.

Reviewer 2: I tested a bit and it seems they indeed aren't 😄

   :alt: Default sales and purchase taxes are set in the Accounting settings.

:guilabel:`Prices` can be changed to :guilabel:`Tax Included` to treat all taxes as :ref:`Tax
Included <taxes/included-in-price>` by default. This would be appropriate if all of a company's
pricing is done tax-included.

.. image:: taxes/tax-prices.png
   :alt: The Tax Prices setting controls whether prices are Tax Excluded or Tax Included by default.

EU intra-community distance selling
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Activate this option if you need to sell to consumers in other EU countries and apply local VAT
rates. More information can be found at the :doc:`EU intra-community distance selling
<taxes/eu_distance_selling>` page.

Cash basis
~~~~~~~~~~

Activate this option if taxes must be accounted for on a cash rather than accruals basis. This might
be the case if the company operates under such a legal regime. Some countries mandate cash basis
accounting, in which case this option will be activated by default by the :doc:`fiscal localization
package <../fiscal_localizations>`. More information can be found at the :doc:`Cash Basis
<taxes/cash_basis>` page.

.. _taxes/list_activation:

> Activate pre-configured taxes

Reviewer: Let's also mention localizations here

Taxes
=====

Tax regulations generally require companies to compute tax amounts on sales, keep a record of
accumulated tax debit and credit, and periodically file this information in tax returns.

Overview
========

In general, managing taxes in Odoo involves the following steps:

1. Taxes are added on individual lines of documents created via the Sale, Purchase, Accounting and
   Point of Sale apps.
2. Odoo automatically computes tax amounts on the documents.
3. On accounting documents, Odoo generates journal items to keep track of tax debit and tax credit.

> 4. The total base, tax debit and tax credit for a period can be viewed in the Tax Return report

Reviewer 3: ```suggestion
4. The total base, tax debit, and tax credit for a period can be viewed in the Tax Return report
```

   and used to file a tax return.

Taxes on sales and purchases
----------------------------

Most sale and purchase documents have a :guilabel:`Taxes` field where taxes can be applied to
individual lines. This includes :doc:`invoices <customer_invoices>` and :doc:`vendor bills
<vendor_bills>` in the :guilabel:`Accounting` app, :doc:`sales quotations
<../../sales/sales/sales_quotations>` in the :guilabel:`Sales` app, and :doc:`purchase orders
<../../inventory_and_mrp/purchase/manage_deals/rfq>` in the :guilabel:`Purchase` app.

.. image:: taxes/invoice-tax.png
   :alt: Adding a 21% tax on an invoice line.

.. tip::
   When adding a product to a sale or purchase line, the taxes :ref:`assigned to the product
   <taxes/product>` are automatically applied on the line.

.. note::

>    In the :doc:`Point of Sale <../../sales/point_of_sale>`, the taxes :ref:`assigned to the products

Reviewer 3: ```suggestion
   In the :doc:`Point of Sale <../../sales/point_of_sale>` app, the taxes :ref:`assigned to the products
```

   <taxes/product>` are applied directly to the order and cannot be changed except by a
   :doc:`fiscal position <taxes/fiscal_positions>`.

Automatic computation of tax amounts
------------------------------------

> Applying a tax on a sale or purchase line lets Odoo automatically compute a tax amount based on

Reviewer 3: ```suggestion
Applying a tax to a sale or purchase line allows Odoo to automatically compute the tax amount based on
```

the sale or purchase line's subtotal and the tax's configuration. The details of the computation are
explained in the :doc:`Tax Computation <taxes/tax_computation>` page.

.. image:: taxes/invoice-tax-amount.png
   :alt: Odoo automatically computes a tax amount for the 21% tax.

Automatic generation of tax journal items
-----------------------------------------

> Upon applying a tax on an :doc:`Invoice <customer_invoices>` or :doc:`Vendor Bill <vendor_bills>`, a

Reviewer 3: ```suggestion
Upon applying a tax on an :doc:`invoice <customer_invoices>` or :doc:`vendor bill <vendor_bills>`, a
```

tax payable journal item is automatically generated with the tax amount. This keeps track of the tax
debit or credit associated with the transaction.

> Furthermore, the tax amount is added to the amount due on the Account Receivable or Payable journal

Reviewer 3: ```suggestion
Furthermore, the tax amount is added to the amount due on the account receivable or payable journal
```
Maybe use italics instead of capitals to distinguish these specific terms?

2. Activate any needed :ref:`pre-configured taxes <taxes/list_activation>`
3. Assign taxes on your :ref:`products <taxes/product>`

.. _taxes/configuration/company:

Company-wide options
--------------------

To access these configuration options, go to :menuselection:`Accounting --> Configuration
--> Settings` and scroll down to :guilabel:`Taxes`.

.. _taxes/default:

Default taxes
~~~~~~~~~~~~~

The default :guilabel:`Sales Tax` and :guilabel:`Purchase Tax` are automatically set on products
when creating new products.

If :ref:`Accounting Firms <accounting/fiduciaries>` mode is enabled, the default sales tax is
automatically set on new invoice lines, and the default purchase tax is automatically set on new
vendor bill lines.

:guilabel:`Prices` can be changed to :guilabel:`Tax Included` to treat all taxes as :ref:`Tax
Included <taxes/included-in-price>` by default. This would be appropriate if all of a company's
pricing is done tax-included. If only some of a company's pricing is tax-included, individual taxes
can be set as :guilabel:`Tax Included`.

EU intra-community distance selling
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Activate this option if you need to sell to consumers in other EU countries and apply local VAT

Reviewer 3: ```suggestion
Activate this option if you are based in the EU and sell to consumers in other EU countries to apply local VAT
```

rates.

.. seealso::
   :doc:`EU intra-community distance selling<taxes/eu_distance_selling>`

Cash basis
~~~~~~~~~~

Activate this option if taxes must be accounted for on a cash rather than accruals basis. This might
be the case if the company operates under such a legal regime. Some countries mandate cash basis
accounting, in which case this option will be activated by default by the :doc:`fiscal localization
package <../fiscal_localizations>`.

.. seealso::

>    :doc:`Cash Basis <taxes/cash_basis>`

Reviewer 3: ```suggestion
   :doc:`Cash basis <taxes/cash_basis>`
```

.. _taxes/list_activation:

Activate pre-configured taxes
-----------------------------

> The list of taxes can be accessed at :menuselection:`Accounting --> Configuration --> Taxes`.

Reviewer 3: ```suggestion
The list of taxes can be accessed by going to :menuselection:`Accounting --> Configuration --> Taxes`.
```

By default, inactive taxes are created for most sales tax rates by the :doc:`fiscal localization
<../fiscal_localizations>` package, but only the main tax rate is active by default. To activate an
inactive tax, click the toggle in the :guilabel:`Active` column.

.. _taxes/product:

Assign taxes on products
------------------------

To configure the taxes used for each product, go to :menuselection:`Accounting --> Customers -->
Products`, select the product to configure, and fill the :guilabel:`Sales Taxes` and
:guilabel:`Purchase Taxes` fields. These taxes are automatically applied when adding the product to

> a sale or purchase.

Reviewer 3: ```suggestion
a sales or purchase order.
```

Tax computation
~~~~~~~~~~~~~~~

The :guilabel:`Tax Computation` field determines how the tax amount is computed from the sales
price. The following options are available:

- :ref:`Group of Taxes <taxes/computation/group-of-taxes>`: a combination of several other taxes
- :ref:`Fixed <taxes/computation/fixed>`: a fixed amount
- :ref:`Percentage of Price <taxes/computation/percentage-of-price>`: a percentage of the
  tax-excluded sales price
- :ref:`Percentage of Price Tax Included <taxes/computation/percentage-of-price-tax-included>`: a
  percentage of the tax-included total

> - :ref:`Custom Formula <taxes/computation/python-code>`: a custom user-defined formula

Reviewer: ```suggestion
- :ref:`Custom Formula <taxes/computation/python-code>`: a custom, user-defined formula
```

Configure how the tax appears to your customers
-----------------------------------------------

.. _taxes/label-invoices:

Label on invoices
~~~~~~~~~~~~~~~~~

The :guilabel:`Label on Invoices` appears on invoice lines in invoice PDFs and on the customer
portal.

.. image:: taxes/invoice-portal-tax.png
   :alt: The Label on Invoice shows on the invoice line when viewed in the customer portal.

.. _taxes/tax-group:

Tax group
~~~~~~~~~

The :guilabel:`Tax Group` is shown in the totals section of the invoice, in invoice PDFs and on the
customer portal. Multiple taxes that belong to the same tax group are aggregated together into a
single tax amount.

.. image:: taxes/invoice-portal-total.png
   :alt: The Tax Group shows in the totals section when viewed in the customer portal.

.. _taxes/definition-tab:

Configure how tax journal items are created
-------------------------------------------

The :guilabel:`Distribution for Invoices` and :guilabel:`Distribution for Refunds` sections control
the generation of Tax Payable journal items in invoices and credit notes, respectively. They also
determine which :guilabel:`Tax Grids` are set on invoice lines when this tax is applied.

> Each of these sections should contain one :guilabel:`Base` line, one :guilabel:`100.00% of tax`

Reviewer 2: Thanks so much, I updated the explanation to include these!!!

The :guilabel:`Base` line can have one or more :ref:`Tax Grids <tax-returns/tax-grids>` set. Those
:guilabel:`Tax Grids` will be added to the invoice lines on which the tax is applied.

The :guilabel:`100.00% of tax` line controls the creation of the Tax Payable journal item with the
tax amount that represents the tax liability or credit associated with the tax. It should specify
the :guilabel:`Account` in which to create the journal item, and can provide one or more
:guilabel:`Tax Grids` to set on the journal item.

Optionally, a :guilabel:`-100.00% of tax` line can be added. In this case, a second Tax Payable or
Tax Receivable journal item will be created with the negative amount of the tax. Use this if a
non-zero tax amount must be computed, but the application of the tax simultaneously generates both a
tax debit and a tax credit which cancel each other out (e.g. EU intra-community reverse-charge VAT).

.. image:: taxes/distribution-invoices.png
   :alt: The Distribution for Invoices of a 21% VAT tax.

.. image:: taxes/distribution-refunds.png
   :alt: The Distribution for Refunds of a 21% VAT tax.

Extra taxes
===========

> `TODO ANDU: Remove this section. I'm leaving it while we consider to what extent the content is

Reviewer: I feel like this information should be kept, and I don't see a better place to put it instead of here. If you have other suggestions, I'm open to them. Otherwise, let's keep it here.

base amount and a tax amount. This makes it suitable for B2C sales in most countries where prices
are quoted tax-inclusive.

.. example::
   A product has a sales price of $1000, and we apply a 10% :guilabel:`Percentage of Price` tax
   with :guilabel:`Included in Price` set to :guilabel:`Tax Included`. We then have:

   +-------------+-------------+----------+----------+
   | Product     | Price       | Tax      | Total    |
   | sales price | without tax |          |          |
   +=============+=============+==========+==========+
   | 1,000       | 909.09      | 90.91    | 1,000.00 |
   +-------------+-------------+----------+----------+

.. note::
   For a guide on configuring tax-excluded and tax-included prices for B2B and B2C customers,
   see :doc:`taxes/B2B_B2C`.

.. _taxes/base-subsequent:

Affect base of subsequent taxes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If this setting is enabled, any subsequent tax applied on the same product line that has
:ref:`taxes/base-affected` will be based on a modified sales price. The exact behavior depends on
the :ref:`Included in Price <taxes/included-in-price>` setting of the tax with :guilabel:`Affect

> base of subsequent taxes`.

Reviewer: I think your suggestion works just fine :) I'd just change the spelling of "behaviour" to "behavior" since we use US spellings, not UK. I still think other sentences in this section could be made more clear by talking about fields and settings being enabled instead of just using the field name as part of the sentence structure itself:

> If **Included in Price** is set to **Tax Excluded**, subsequent taxes with **Base affected by preceding taxes** will be based on a modified sales price equal to the original sales price plus the tax amount.

would be more clear as:

> If **Included in Price** is set to **Tax Excluded**, subsequent taxes with the **Base affected by preceding taxes** setting enabled will be based on a modified sales price equal to the original sales price plus the tax amount.

.. tabs::
   .. tab:: Tax-excluded
      If :guilabel:`Included in Price` is set to :guilabel:`Tax Excluded`, subsequent taxes with
      :guilabel:`Base affected by preceding taxes` will be based on a modified sales price equal to
      the original sales price plus the tax amount.

      .. example::
         A product has a sales price of $1000, and we apply a 10% :guilabel:`Percentage of Price`
         tax with :guilabel:`Affect base of subsequent taxes`. Any subsequent tax with
         :guilabel:`Base affected by preceding taxes` will be based on a modified sales price of
         $1100.

   .. tab:: Tax-included
      If :guilabel:`Included in Price` is set to :guilabel:`Tax Included`, subsequent taxes with
      :guilabel:`Base affected by preceding taxes` will be based on a modified sales price equal to
      the original sales price minus the tax amount.

      .. example::
         A product has a sales price of $1100, and we apply a 10% :guilabel:`Percentage of Price`
         tax that has :guilabel:`Tax Included` and :guilabel:`Affect base of subsequent taxes`.
         Any subsequent tax with :guilabel:`Base affected by preceding taxes` will be based on a

>          modified sales price of $1000.

Reviewer: So it uses the tax-free amount here?? That seems strange.

Reviewer 2: It does work this way - I did a runbot test to double-check. I created two 10% tax-included taxes with `Affect base of subsequent taxes`, and an invoice for $110. The tax amounts show that the second tax is computed on a base of $110, and the first tax is computed on a base of $100.

<img width="838" height="368" alt="image" src="https://github.com/user-attachments/assets/b2d44068-dd62-447b-a3a2-f44d4419418b" />

Another interesting thing to observe is that although the second tax is later in sequence than the first, it is actually computed first. I had a quick look at the code and found that where several tax-included taxes are applied, their order of computation is reversed, so that from the perspective of the untaxed amount, the first tax is innermost relative to the second. Would you think this merits being documented, or whether it would be too much detail?

         the :guilabel:`Affect base of subsequent taxes` setting is enabled. Any subsequent tax with
         its :guilabel:`Base affected by preceding taxes` will be based on a modified sales price of
         $1100.

   .. tab:: Tax-included
      If :guilabel:`Included in Price` is set to :guilabel:`Tax Included`, subsequent taxes with
      :guilabel:`Base affected by preceding taxes` will be based on a modified sales price equal to
      the original sales price minus the tax amount.

      .. example::
         A product has a sales price of $1100, and we apply a 10% :guilabel:`Percentage of Price`
         tax where the :guilabel:`Included in Price` setting is set to :guilabel:`Tax Included` and
         the :guilabel:`Affect base of subsequent taxes` setting is enabled. Any subsequent tax with
         its :guilabel:`Base affected by preceding taxes` will be based on a modified sales price of
         $1000.

This setting is considered any time multiple taxes are applied to the same product line, whether
via a :ref:`group of taxes <taxes/computation>` or multiple taxes added directly to a product line.

.. warning::

>    The order in which taxes are applied depends only on the order in which they appear in the

Reviewer 2: I think the order in which the taxes are applied by the engine might be a bit unhelpful to the user. It is true that technically, that's the order in which the computations occur. But the engine also tries to put the taxes back in the 'right order' afterwards.

Anyways, the practical consequence of the evaluation order is that Tax Excluded taxes cannot affect the base of subsequent Tax Included taxes. In fact, I suspect this is the reason why the `Is Base Affected` setting doesn't appear on Tax Included taxes (though I didn't double-check this with Laurent). Tax Included taxes thus always get affected by other preceding Tax Included taxes which have `Affect base of subsequent taxes`, and never get affected by preceding Tax Excluded taxes.

I wrote about this in the `Is Base Affected` section and also added a small word here:

"Regardless of the order in the :guilabel:`Taxes` list, :guilabel:`Tax Excluded` taxes do not affect the base of subsequent :guilabel:`Tax Included` taxes (see the note in :ref:`taxes/base-affected`)".

Let me know what you think of this!

To compute an extra tax in Odoo, :ref:`create a tax <taxes/configuration>`, enter a tax name, select
a :ref:`Tax Computation <taxes/configuration>`, set an :guilabel:`Amount`, and in the
:guilabel:`Advanced Options` tab, enable :guilabel:`Affect Base of Subsequent Taxes`. Then, drag and

> drop the taxes in the :ref:`order they should be computed <taxes/base-subsequent>`.

Reviewer 2: I think we could just keep the current meaning of this sentence for now.
